qiling/extensions/r2/r2.py

```diff
@@ -102,14 +103,35 @@ class SymbolBind(str, Enum):
 
 
 class R2:
-    def __init__(self, ql: Qiling):
+    def __init__(self, ql: Qiling, baseaddr=(1 << 64) - 1, loadaddr=0):
         super().__init__()
-        path = ql.path.encode()
-        self._r2c = libr.r_core.r_core_new()
-        fh = libr.r_core.r_core_file_open(self._r2c, path, 0b101, 0)
-        libr.r_core.r_core_bin_load(self._r2c, path, (1 << 64) - 1)
+        self.ql = ql
+        self.baseaddr = baseaddr  # r2 -B [baddr]   set base address for PIE binaries
+        self.loadaddr = loadaddr  # r2 -m [addr]    map file at given address
 
-    def _cmd(self, cmd: str):
+        self._r2c = libr.r_core.r_core_new()
+        if ql.code:
+            self._setup_code()
+        else:
+            self._setup_file()
+
+        # set architecture and bits for r2 asm
+        self._cmd(f"e,asm.arch={ql.arch.type.name.lower().removesuffix('64')},asm.bits={ql.arch.bits}")
+
+    def _setup_code(self):
+        sz = len(self.ql.code)
+        path = f'malloc://{sz}'.encode()
+        fh = libr.r_core.r_core_file_open(self._r2c, path, UC_PROT_ALL, self.loadaddr)
+        libr.r_core.r_core_bin_load(self._r2c, path, self.baseaddr)
+        cmd = f'wx {self.ql.code.hex()}'
+        self._cmd(cmd)
+
+    def _setup_file(self):
+        path = self.ql.path.encode()
+        fh = libr.r_core.r_core_file_open(self._r2c, path, UC_PROT_READ | UC_PROT_EXEC, self.loadaddr)
+        libr.r_core.r_core_bin_load(self._r2c, path, self.baseaddr)
+
+    def _cmd(self, cmd: str) -> str:
```

Comment:
Consider adding a matching `cmdj` method to handle `j`-suffixed commands, instead of calling `json.loads` every time.